From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 7 Sep 2023 18:00:21 +0200
Subject: screenshot: Only handle mode-switch shortcut when supported

We currently handle the 'v' key to switch between recording- and
screenshot mode regardless of whether screen recordings are
supported.

This is clearly wrong, don't do that.

(cherry picked from commit 671df28a509ae208e158976f0855d91fdbea16a1)

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/6990
CVE: CVE-2023-43090
Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2944
Applied-upstream: 43.9, commit:673e78457257243d408c85285eb8324f7954e395
Bug-Debian: https://bugs.debian.org/1052067
---
 js/ui/screenshot.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/ui/screenshot.js b/js/ui/screenshot.js
index 84830b2..5139052 100644
--- a/js/ui/screenshot.js
+++ b/js/ui/screenshot.js
@@ -2001,7 +2001,8 @@ var ScreenshotUI = GObject.registerClass({
             return Clutter.EVENT_STOP;
         }
 
-        if (symbol === Clutter.KEY_v || symbol === Clutter.KEY_V) {
+        if (this._castButton.reactive &&
+            (symbol === Clutter.KEY_v || symbol === Clutter.KEY_V)) {
             this._castButton.checked = !this._castButton.checked;
             return Clutter.EVENT_STOP;
         }
